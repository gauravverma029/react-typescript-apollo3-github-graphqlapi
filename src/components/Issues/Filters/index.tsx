import React, { useState } from "react";

type Props = {
  handleFilterStates: (props: FiterStateProps) => void;
};

const Filters: React.FC<Props> = ({ handleFilterStates }) => {
  const [isActive, setIsActive] = useState("");

  const onSelectFilter = ({ type, value }: FiterStateProps) => {
    handleFilterStates({ type, value });
    setIsActive(value);
  };

  return (
    <div className="px-4 py-3 flex bg-gray-100">
      <div className="flex-auto flex space-x-6">
        <div
          className={`flex items-center justify-center  cursor-pointer ${
            isActive === "" ? "text-gray-400" : "text-gray-600"
          }`}
          onClick={() => onSelectFilter({ type: "state", value: "" })}
        >
          All
        </div>
        <div
          className={`flex items-center justify-center cursor-pointer  ${
            isActive === "is:open" ? "text-gray-400" : "text-gray-600"
          }`}
          onClick={() => onSelectFilter({ type: "state", value: "is:open" })}
        >
          Open
        </div>
        <div
          className={`flex items-center justify-center cursor-pointer ${
            isActive === "is:closed" ? "text-gray-400" : "text-gray-600"
          }`}
          onClick={() => onSelectFilter({ type: "state", value: "is:closed" })}
        >
          Closed
        </div>
      </div>
    </div>
  );
};
export default Filters;
