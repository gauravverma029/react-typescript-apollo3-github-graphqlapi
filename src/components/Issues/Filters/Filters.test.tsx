import React from "react";
import { cleanup, render, fireEvent } from "@testing-library/react";

import Filters from "./index";

afterEach(cleanup);

const handleFilterStates = () => {};

describe("Filter Component", () => {
  test("Should Render", async () => {
    const { asFragment } = render(
      <Filters handleFilterStates={handleFilterStates} />
    );
    const html = asFragment();
    expect(html).toMatchSnapshot();
  });

  test("Should Contain Filter All", async () => {
    const { getByText } = render(
      <Filters handleFilterStates={handleFilterStates} />
    );
    getByText("All");
  });

  test("Should Contain Filter Open", async () => {
    const { getByText } = render(
      <Filters handleFilterStates={handleFilterStates} />
    );
    getByText("Open");
  });

  test("Should Contain Filter Closed", async () => {
    const { getByText } = render(
      <Filters handleFilterStates={handleFilterStates} />
    );
    getByText("Closed");
  });

  test("Click on All Filter", async () => {
    const { getByText } = render(
      <Filters handleFilterStates={handleFilterStates} />
    );
    fireEvent.click(getByText("All"));
  });
  test("Click on Open Filter", async () => {
    const { getByText } = render(
      <Filters handleFilterStates={handleFilterStates} />
    );
    fireEvent.click(getByText("Open"));
  });
  test("Click on Closed Filter", async () => {
    const { getByText } = render(
      <Filters handleFilterStates={handleFilterStates} />
    );
    fireEvent.click(getByText("Closed"));
  });
});
