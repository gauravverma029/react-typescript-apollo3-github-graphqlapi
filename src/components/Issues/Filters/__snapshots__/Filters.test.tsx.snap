// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component Should Render 1`] = `
<DocumentFragment>
  <div
    class="px-4 py-3 flex bg-gray-100"
  >
    <div
      class="flex-auto flex space-x-6"
    >
      <div
        class="flex items-center justify-center  cursor-pointer text-gray-400"
      >
        All
      </div>
      <div
        class="flex items-center justify-center cursor-pointer  text-gray-600"
      >
        Open
      </div>
      <div
        class="flex items-center justify-center cursor-pointer text-gray-600"
      >
        Closed
      </div>
    </div>
  </div>
</DocumentFragment>
`;
